helpers/helper_entitymatching.go

```diff
@@ -0,0 +1,141 @@
+	resp, err := c.ClientConfig.CreateConfigNode(ctx, createReq)
+	if err != nil {
+		return nil, errors.NewInvalidArgumentErrorWithCause(err, "failed to Create ConfigNode on IndyKite Client")
+	}
+	idPipeline := resp.Id
+
+	// call ReadSuggestedPropertyMapping with interval until success
+	if !readSuggestedPropertyMappingWithInterval(ctx, c, idPipeline, duration) {
+		return nil, errors.NewInvalidArgumentErrorWithCause(
+			err,
+			"failed to run readSuggestedPropertyMappingWithInterval on IndyKite Client")
+	}
+
+	// run RunEntityMatchingPipeline
+	reqRun := &entitymatchingpb.RunEntityMatchingPipelineRequest{
+		Id:                    idPipeline,
+		SimilarityScoreCutoff: similarityScoreCutoff,
+	}
+	respRun, err := c.ClientEntitymatching.RunEntityMatchingPipeline(context.Background(), reqRun)
+	if err != nil {
+		return nil, errors.NewInvalidArgumentErrorWithCause(
+			err,
+			"failed to run RunEntityMatchingPipeline on IndyKite Client")
+	}
+
+	if respRun.Id != idPipeline {
+		return nil, errors.NewInvalidArgumentError(
+			"failed to retrieve id from  RunEntityMatchingPipeline on IndyKite Client")
+	}
+
+	// call ReadConfigNode with interval until success
+	if !readConfigNodeWithInterval(ctx, c, idPipeline, duration) {
+		return nil, errors.NewInvalidArgumentErrorWithCause(
+			err,
+			"failed to run readConfigNodeWithInterval on IndyKite Client")
+	}
+	return &CreateAndRun{configNodeID: idPipeline}, nil
+}
+
+func readSuggestedPropertyMappingWithInterval(ctx context.Context, c *Client, id string, interval time.Duration) bool {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return false
+		case <-ticker.C:
+			reqPropertyMapping := &entitymatchingpb.ReadSuggestedPropertyMappingRequest{
+				Id: id,
+			}
+			//nolint:contextcheck //the context is different
+			respPropertyMapping, err := c.ClientEntitymatching.ReadSuggestedPropertyMapping(
+				context.Background(),
+				reqPropertyMapping)
+			if err != nil {
+				return false
+			}
+			if respPropertyMapping.PropertyMappingStatus ==
+				entitymatchingpb.PipelineStatus_PIPELINE_STATUS_STATUS_SUCCESS {
+				return true
+			}
+			return true
+		}
+	}
+}
+
+func readConfigNodeWithInterval(ctx context.Context, c *Client, id string, interval time.Duration) bool {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return false
+		case <-ticker.C:
+			configNodeRequest, err := config.NewRead(id)
+			if err != nil {
+				return false
+			}
+			//nolint:contextcheck //the context is different
+			configNodeResponse, err := c.ClientConfig.ReadConfigNode(
+				context.Background(),
+				configNodeRequest)
+			if err != nil {
+				return false
+			}
+			if configNodeResponse.ConfigNode.GetEntityMatchingPipelineConfig().EntityMatchingStatus ==
+				configpb.EntityMatchingPipelineConfig_STATUS_SUCCESS {
+				return true
+			}
+		}
+	}
+}
```

Comment:
Suggestion: if we keep the logic probably the polling logic can be extracted as a separate function which takes an argument of type function, to avoid duplication.

Reply:
you are absolutely right: I will remove the duplicates in next version